contrib/devtools/security-check.py

```diff
@@ -111,6 +111,17 @@ def check_ELF_separate_code(binary):
                 return False
     return True
 
+def check_ELF_control_flow(binary) -> bool:
+    '''
+    Check for control flow instrumentation
+    '''
+    main = binary.get_function_address('main')
+    content = binary.get_content_from_virtual_address(main, 4, lief.Binary.VA_TYPES.AUTO)
+
+    if content == [243, 15, 30, 250]: # endbr64
```

Comment:
This feels kind of ugly. Is there no better way to check it? Will this reliably check it with future compilers?

Reply:
> Is there no better way to check it?

Do you have any suggestions? As far as I'm aware there's no standard way to check if a binary has control instrumentation or not. I think this is about the most direct way we can test for what we're looking for.

> Will this reliably check it with future compilers?

I'm not going to try and guess what future compilers may or may not do, but I would be surprised if this did stop working. In any case, it would be immediately obvious, and would only occur when making changes to our Guix environment, so at that point we could figure out what to do.

Reply:
I think this is the only way to do it. I also think it is future-proof, there's only one way to do control flow experimentation. And yeah, if not, we can check what to do. Compiler upgrades are an explicit choice anyhow.